klippy/extras/servo.py

```diff
@@ -28,28 +28,51 @@ def __init__(self, config):
         self.gcode.register_mux_command("SET_SERVO", "SERVO", servo_name,
                                         self.cmd_SET_SERVO,
                                         desc=self.cmd_SET_SERVO_help)
-    def set_pwm(self, print_time, value):
+
+        self.initial_pwm_value = -1
+        #Check to see if an initial angle or pulse width is configured and
+        # set it as required
+        initial_angle = config.getfloat('initial_angle', -1)
+        if initial_angle != -1:
+            self.initial_pwm_value = round(self._get_pwm_from_angle(initial_angle),3)
+        else:
+            initial_pulse_width = config.getfloat('initial_pulse_width', -1)
+            if initial_pulse_width != -1:
+                self.initial_pwm_value = round(self._get_pwm_from_pulse_width(
+                    initial_pulse_width),3)
+
+    def printer_state(self, state):
+        if state == 'ready':
+            print_time = self.printer.lookup_object('toolhead').get_last_move_time()
+            self._set_pwm(print_time, self.initial_pwm_value)
```

Comment:
It's not clear to me if this is correct when an initial value is not set in the config.  I'd say only run set_pwm() if the user has specified an initial value.

Reply:
Fixed up in the latest commit